orchagent/vxlanorch.cpp

```diff
@@ -515,19 +631,544 @@ bool VxlanTunnel::removeNextHop(IpAddress& ipAddr, MacAddress macAddress, uint32
     {
         if (sai_next_hop_api->remove_next_hop(nh_tunnels_[key].nh_id) != SAI_STATUS_SUCCESS)
         {
+            SWSS_LOG_NOTICE("delete NH tunnel for ip '%s', mac '%s' vni %d failed",
+                            ipAddr.to_string().c_str(), macAddress.to_string().c_str(), vni);
             string err_msg = "NH tunnel delete failed for " + ipAddr.to_string();
             throw std::runtime_error(err_msg);
         }
 
         nh_tunnels_.erase(key);
     }
 
-    SWSS_LOG_INFO("NH tunnel for ip '%s', mac '%s' updated/deleted",
-                   ipAddr.to_string().c_str(), macAddress.to_string().c_str());
+    SWSS_LOG_NOTICE("NH tunnel for ip '%s', mac '%s' vni %d updated/deleted",
+                    ipAddr.to_string().c_str(), macAddress.to_string().c_str(), vni);
+
+    return true;
+}
+
+bool VxlanTunnel::deleteMapperHW(uint8_t mapper_list,
+                                 tunnel_map_src_t map_src)
+{
+    sai_status_t status = SAI_STATUS_SUCCESS;
+
+    SWSS_LOG_INFO("Enter deleteMapper HW maplst=%d src=%d",mapper_list, map_src);
+
+    try
+    {
+      if(map_src == USE_DEDICATED_ENCAP_DECAP)
+      {
+        if(IS_TUNNELMAP_SET_VLAN(mapper_list))
```

Comment:
There is a bunch of duplicated code. can we optimize it to and just modify the index in the` if `case

Reply:
Encap/Decap mappers, 
Different mapper types - VRF/VLAN/BRIDGE,
Different use cases - VTEP, P2P, DCI tunnels .. 
All these have to be handled. Optimised to the best possible extent.